odiglet/pkg/ebpf/configprovider.go

```diff
@@ -0,0 +1,69 @@
+package ebpf
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/odigos-io/odigos/odiglet/pkg/log"
+)
+
+type ConfigProvider[C any] struct {
+	configChan    chan C
+	initialConfig C
+
+	stoppedMutex sync.Mutex
+	stopped      bool
+}
+
+// NewConfigProvider creates a new configProvider with the given initial config.
+// It allows for updating the configuration of a running instrumentation.
+func NewConfigProvider[C any](initialConfig C) *ConfigProvider[C] {
+	return &ConfigProvider[C]{
+		initialConfig: initialConfig,
+		configChan:    make(chan C),
+	}
+}
+
+func (c *ConfigProvider[C]) InitialConfig(_ context.Context) C {
+	return c.initialConfig
+}
+
+func (c *ConfigProvider[C]) Shutdown(_ context.Context) error {
+	c.stoppedMutex.Lock()
+	defer c.stoppedMutex.Unlock()
+
+	if c.stopped {
+		return nil
+	}
+
+	close(c.configChan)
+	c.stopped = true
+	return nil
+}
+
+func (c *ConfigProvider[C]) Watch() <-chan C {
```

Comment:
Shouldn't we pass `context.Context` here as well? And maybe cancel it when we want to stop watching?

Reply:
I'm not sure since it is ment to be a getter for the channel. 
The calling code in the instrumentation looks like:
```
	for {
		select {
		case <-ctx.Done():
			return
		case c, ok := <-m.cp.Watch():
			if !ok {
				m.logger.Info("Configuration provider closed, configuration updates will no longer be received")
				return
			}
			err := m.applyConfig(c)
			if err != nil {
				m.logger.Error(err, "Failed to apply config")
				continue
			}
			m.currentConfig = c
		}
	}
```
The way to stop watching should be to call `Shutdown` - which is called by the instrumentation once it is closing